components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/valve/TenantContextRewriteValve.java

```diff
@@ -273,17 +280,47 @@ private void handleInvalidTenantDomainErrorResponse(Response response, int error
         response.getWriter().print(errorResponse.toString());
     }
 
-    private void handleRestrictedTenantDomainErrorResponse(Response response) throws IOException, ServletException {
+    private void handleRestrictedTenantDomainErrorResponse(Request request, Response response) throws IOException {
+
+        String requestContentType = request.getContentType();
+        if (StringUtils.contains(requestContentType, "application/json")) {
+            response.setContentType("application/json");
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+            response.setCharacterEncoding("UTF-8");
+            JsonObject errorResponse = new JsonObject();
+            errorResponse.addProperty("code", HttpServletResponse.SC_FORBIDDEN);
+            String errorMsg = "Access to super tenant domain over tenanted URL format (/t/carbon.super) is restricted. "
+                    + "Please use the server base path instead.";
+            errorResponse.addProperty("message", errorMsg);
+            errorResponse.addProperty("description", errorMsg);
+            response.getWriter().print(errorResponse.toString());
+        } else {
+            response.setContentType("text/html");
+            String errorPage = loadPageNotFoundErrorPage();
+            response.getWriter().print(errorPage);
+        }
+    }
 
-        response.setContentType("application/json");
-        response.setStatus(HttpServletResponse.SC_FORBIDDEN);
-        response.setCharacterEncoding("UTF-8");
-        JsonObject errorResponse = new JsonObject();
-        String errorMsg = "Access to super tenant domain over tenanted URL format (/t/carbon.super) is restricted. " +
-                "Please use the server base path instead.";
-        errorResponse.addProperty("code", HttpServletResponse.SC_FORBIDDEN);
-        errorResponse.addProperty("message", errorMsg);
-        errorResponse.addProperty("description", errorMsg);
-        response.getWriter().print(errorResponse.toString());
+    private String loadPageNotFoundErrorPage() {
+
+        String errorPage = "Page Not Found";
+        try {
+            Path pageNotFoundHtmlResponse =
+                    Paths.get(CarbonUtils.getCarbonHome(), "repository", "resources", "identity", "pages",
```

Comment:
It is better if we can load this in the component activate method and resolve the error page. Then reuse other than checking or each API calls.

Reply:
Fixed with https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/145/commits/a6ec2c2a744b12df63be45d834e8b7ec0cdafb8f